lib/kino_proxy/server.ex

```diff
@@ -0,0 +1,40 @@
+defmodule KinoProxy.Server do
+  # TODO: Add doc comments
+  @moduledoc false
+
+  @proxy_params ["id", "path"]
+
+  def run(pid, %Plug.Conn{params: %{"path" => path_info}} = conn) when is_pid(pid) do
+    {mod, state} = conn.adapter
+    %{:plug_session => session_data} = conn.private
+    request_path = "/" <> Enum.join(path_info, "/")
+    private = %{plug_session: session_data}
+    params = Map.drop(conn.params, @proxy_params)
+    path_params = Map.drop(conn.path_params, @proxy_params)
+
+    conn = %{
+      conn
+      | request_path: request_path,
+        path_info: path_info,
+        params: params,
+        path_params: path_params,
+        private: private
+    }
+
+    spawn_pid = GenServer.call(pid, {:request, conn, self()})
+    monitor_ref = Process.monitor(spawn_pid)
+    loop(spawn_pid, monitor_ref, mod, state)
+  end
+
+  defp loop(spawn_pid, monitor_ref, mod, state) do
+    receive do
+      {:send_resp, pid, ref, args} ->
+        {:ok, body, state} = apply(mod, :send_resp, [state | args])
+        send(pid, {ref, {:ok, body, self()}})
+        loop(spawn_pid, monitor_ref, mod, state)
+
+      {:DOWN, ^monitor_ref, :process, ^spawn_pid, _} ->
+        loop(spawn_pid, monitor_ref, mod, state)
```

Comment:
You are supposed to return here. :) Something like `:ok` but perhaps `reason` is enough. This should fix the tests.

```suggestion
      {:DOWN, ^monitor_ref, :process, ^spawn_pid, reason} ->
        reason
```

Reply:
In that case, the test always show an error which I'm not sure we should have.

```
11:03:38.466 [error] ** (Plug.Conn.NotSentError) a response was neither set nor sent from the connection
    (bandit 1.5.0) lib/bandit/pipeline.ex:167: Bandit.Pipeline.commit_response!/1
    (bandit 1.5.0) lib/bandit/pipeline.ex:40: Bandit.Pipeline.run/4
    (bandit 1.5.0) lib/bandit/http1/handler.ex:12: Bandit.HTTP1.Handler.handle_data/3
    (bandit 1.5.0) lib/bandit/delegating_handler.ex:18: Bandit.DelegatingHandler.handle_data/3
    (bandit 1.5.0) /home/aledsz/github/livebook-dev/kino_proxy/deps/thousand_island/lib/thousand_island/handler.ex:411: Bandit.DelegatingHandler.handle_continue/2
    (stdlib 5.2.1) gen_server.erl:1085: :gen_server.try_handle_continue/3
    (stdlib 5.2.1) gen_server.erl:995: :gen_server.loop/7
    (stdlib 5.2.1) proc_lib.erl:241: :proc_lib.init_p_do_apply/3
```